github/util_v4_branch_protection.go

```diff
@@ -0,0 +1,271 @@
+	data := BranchProtectionResourceData{}
+
+	if v, ok := d.GetOk(REPOSITORY_ID); ok {
+		data.RepositoryID = v.(string)
+	}
+
+	if v, ok := d.GetOk(PROTECTION_PATTERN); ok {
+		data.Pattern = v.(string)
+	}
+
+	if v, ok := d.GetOk(PROTECTION_IS_ADMIN_ENFORCED); ok {
+		data.IsAdminEnforced = v.(bool)
+	}
+
+	if v, ok := d.GetOk(PROTECTION_REQUIRES_COMMIT_SIGNATURES); ok {
+		data.RequiresCommitSignatures = v.(bool)
+	}
+
+	if v, ok := d.GetOk(PROTECTION_REQUIRES_APPROVING_REVIEWS); ok {
+		vL := v.([]interface{})
+		if len(vL) > 1 {
+			return BranchProtectionResourceData{},
+				fmt.Errorf("error multiple %s declarations", PROTECTION_REQUIRES_APPROVING_REVIEWS)
+		}
+		for _, v := range vL {
+			if v == nil {
+				break
+			}
+
+			data.RequiresApprovingReviews = true
+
+			m := v.(map[string]interface{})
+			if v, ok := m[PROTECTION_REQUIRED_APPROVING_REVIEW_COUNT]; ok {
+				data.RequiredApprovingReviewCount = v.(int)
+			}
+			if v, ok := m[PROTECTION_DISMISSES_STALE_REVIEWS]; ok {
+				data.DismissesStaleReviews = v.(bool)
+			}
+			if v, ok := m[PROTECTION_REQUIRES_CODE_OWNER_REVIEWS]; ok {
+				data.RequiresCodeOwnerReviews = v.(bool)
+			}
+			if v, ok := m[PROTECTION_RESTRICTS_REVIEW_DISMISSALS]; ok {
+				reviewDismissalActorIDs := make([]string, 0)
+				vL := v.(*schema.Set).List()
+				for _, v := range vL {
+					reviewDismissalActorIDs = append(reviewDismissalActorIDs, v.(string))
+				}
+				if len(reviewDismissalActorIDs) > 0 {
+					data.ReviewDismissalActorIDs = reviewDismissalActorIDs
+					data.RestrictsReviewDismissals = true
+				}
+			}
+		}
+	}
+
+	if v, ok := d.GetOk(PROTECTION_REQUIRES_STATUS_CHECKS); ok {
+		vL := v.([]interface{})
+		if len(vL) > 1 {
+			return BranchProtectionResourceData{},
+				fmt.Errorf("error multiple %s declarations", PROTECTION_REQUIRES_STATUS_CHECKS)
+		}
+		for _, v := range vL {
+			if v == nil {
+				break
+			}
+
+			m := v.(map[string]interface{})
+			if v, ok := m[PROTECTION_REQUIRES_STRICT_STATUS_CHECKS]; ok {
+				data.RequiresStrictStatusChecks = v.(bool)
+			}
+
+			data.RequiredStatusCheckContexts = expandNestedSet(m, PROTECTION_REQUIRED_STATUS_CHECK_CONTEXTS)
+			if len(data.RequiredStatusCheckContexts) > 0 {
+				data.RequiresStatusChecks = true
+			}
+		}
+	}
+
+	if v, ok := d.GetOk(PROTECTION_RESTRICTS_PUSHES); ok {
+		pushActorIDs := make([]string, 0)
+		vL := v.(*schema.Set).List()
+		for _, v := range vL {
+			pushActorIDs = append(pushActorIDs, v.(string))
+		}
+		if len(pushActorIDs) > 0 {
+			data.PushActorIDs = pushActorIDs
+			data.RestrictsPushes = true
+		}
+	}
+
+	return data, nil
+}
+
+func setActorIDs(actors []ActorTypes) []string {
+	pushActors := make([]string, 0, len(actors))
+	for _, a := range actors {
+		if a.Actor.Team != (Actor{}) {
+			pushActors = append(pushActors, a.Actor.Team.ID.(string))
+		}
+		if a.Actor.User != (Actor{}) {
+			pushActors = append(pushActors, a.Actor.Team.ID.(string))
```

Comment:
Is this a bug? Should it instead be:
```suggestion
			pushActors = append(pushActors, a.Actor.User.ID.(string))
```
?

Reply:
Will be rectified in https://github.com/integrations/terraform-provider-github/pull/819

Reply:
I've just seen your comment—there's a more complete fix (since the copy-paste error propagated) in #824.